### PR TITLE
Cluster-Kfold. Now with correct spelling!

### PR DESCRIPTION
**Pull request by koaning**
WIP: fixes https://github.com/koaning/scikit-lego/issues/630.

### CODE REVIEW

**Comment by koaning** [comment]
@FBruzzesi let me know if this feels good.

File: sklego/model_selection.py [hunk 1]
```diff
@@ -248,8 +248,17 @@ def get_split_info(X, indices, j, part, summary):
...
+    warn(
+        "Please use `ClusterFoldValidation` instead of `KlusterFoldValidation`."
+        "We will use correct spelling going forward and `KlusterFoldValidation` will be deprecated.",
+        DeprecationWarning,
+    )
+    return ClusterFoldValidation(**kwargs)
+
+
+class ClusterFoldValidation:
+    """Cross validator that create folds based on provided cluster method.
```
Comment (FBruzzesi) [inline sklego/model_selection.py:261]: Couple of addition I can think of are:
- Adding a future release version 
- Extend the description a tiny bit

```suggestion
class ClusterFoldValidation:
    """Cross validator that creates folds based on provided cluster method. Each fold train/test indices are generated 
    by the clustering algorithm. This ensures that data points in the same cluster are not split across different folds.
    
    !!! info "New in version 0.9.0"
```

**Review by FBruzzesi — COMMENTED** [review]
Looks good! I only left a comment for the docstring and versioning.
Let me know if that's reasonable, otherwise we merge it as is

**Review by FBruzzesi — APPROVED** [review]
LGTM ✅

**Pull request merged**